src/MaxMind/Db/Reader/Decoder.php

```diff
@@ -286,17 +286,13 @@ private function decodePointer(int $ctrlByte, int $offset): array
                 // machines
                 $pointerOffset = $this->decodeUint($buffer, $pointerSize);
 
-                $byteLength = $pointerSize + $this->pointerBaseByteSize;
+                $pointerBase = $this->pointerBase;
 
-                if ($byteLength <= _MM_MAX_INT_BYTES) {
+                if (\PHP_INT_MAX - $pointerBase >= $pointerSize) {
                     $pointer = $pointerOffset + $this->pointerBase;
-                } elseif (\extension_loaded('gmp')) {
-                    $pointer = gmp_strval(gmp_add($pointerOffset, $this->pointerBase));
-                } elseif (\extension_loaded('bcmath')) {
-                    $pointer = bcadd($pointerOffset, (string) $this->pointerBase);
                 } else {
                     throw new RuntimeException(
-                        'The gmp or bcmath extension must be installed to read this database.'
+                        'The database offset is too large to be represented on you platform.'
```

Comment:
Wouldn't this be an invalid database? Or I actually given pointer base I suppose one could be constructed that is valid. Huh.

Also: "your platform"

Reply:
I don't think the database would be invalid. We are just hitting the 2 GB file size limit. I believe there are ways around this, but given that our public databases are much smaller and 32-bit platforms are niche, I don't know that it is worth worrying about.